spec/dds_md5_reporter_spec.rb

```diff
@@ -164,42 +227,291 @@
         end
       end
 
-      describe 'called' do
+      shared_context 'auth_token request with error' do
+        let(:expected_code) { "404" }
+        let(:expected_error) { StandardError.new("expected error") }
         include_context 'mocked auth token request'
 
-        context 'for the first time' do
-          it 'should return an auth_token' do
-            expect(subject.auth_token).to eq(expected_api_token)
-          end
+        before do
+          is_expected.to receive(:raise_dds_api_exception)
+            .with("unable to get agent api_token", expected_response)
+            .and_raise(expected_error)
         end
+      end
 
-        context 'when token is not expired' do
-          before do
-            subject.auth_token
-          end
+      shared_context 'successfull auth_token request' do
+        let(:expected_code) { "201" }
+        include_context 'mocked auth token request'
+
+        let(:expected_time_to_live) { 7200 }
+        let(:expected_api_token_response) {
+          {
+            "api_token" => expected_api_token,
+            "time_to_live" => expected_time_to_live
+          }
+        }
+
+        before do
+          expect(expected_response).to receive(:parsed_response)
+            .exactly(expected_calls).times
+            .and_return(expected_api_token_response)
+        end
+      end
+
+      describe 'behavior' do
+        context 'with incorrect credentials' do
+          include_context 'auth_token request with error'
 
           it {
-            expect(subject.auth_token).to eq(expected_api_token)
+            expect {
+              subject.auth_token
+            }.to raise_error(expected_error)
           }
         end
 
-        context 'when token is expired' do
-          let(:expected_calls) { 2 }
-          before do
-            subject.auth_token
+        context 'with correct credentials' do
+          include_context 'successfull auth_token request'
+
+          context 'for the first time' do
+            it 'should return an auth_token' do
+              expect(subject.auth_token).to eq(expected_api_token)
+            end
           end
 
-          it {
-            travel_to(Time.now + 10000) do
+          context 'when token is not expired' do
+            before do
+              subject.auth_token
+            end
+
+            it {
               expect(subject.auth_token).to eq(expected_api_token)
+            }
+          end
+
+          context 'when token is expired' do
+            let(:expected_calls) { 2 }
+            before do
+              subject.auth_token
             end
+
+            it {
+              travel_to(Time.now + 10000) do
+                expect(subject.auth_token).to eq(expected_api_token)
```

Comment:
I'm finding the `#auth_token` tests difficult to follow, this one in particular. Please add descriptions to the `it` blocks.

Reply:
I will refactor this test. I really like using travel_to, but it isnt really apparent that the test expects the auth_token to be refreshed with a new call to the dds api when the token is expired.